File: Interpreted/coin-sums/csum.py
def num_paths(startsum,lastcoin):
    if (startsum==250): return 1
    paths = 0
    for coin in [1,2,5,10,20,50,100,200]:
        if ((lastcoin>=coin) and (startsum<=(250-coin))):
            paths += num_paths(startsum+coin,coin)

    return paths

print(num_paths(0,250))

File: Interpreted/coin-sums/csum.php
<?php
function num_paths($startsum, $lastcoin) {

    static $coins = array(1, 2, 5, 10, 20, 50, 100, 200);

    if ($startsum == 250) {
        return 1;
    }

    $paths = 0;
    foreach ($coins as $coin) {
        if (($lastcoin >= $coin) && ($startsum <= (250 - $coin))) {
            $paths += num_paths($startsum + $coin, $coin);
        }
    }

    return $paths;
}

echo num_paths($argv[1], $argv[2]), "\n";
?>
